api-spec/api-spec.md

```diff
@@ -50,3 +59,198 @@ The `/stac/search` endpoint is similar to the `items` endpoint in WFS3 in that i
+
+## Filter JSON Body
+
+Used with `POST Content-Header: application/json`
+
+All Extensions are **recommended** to use attribute names qualified from the root of Item, rather than Item Properties.
+
+Like Filter Parameters, these attribute names are reserved for extensions, must not implement different semantics.
+
+**TBD** how to allow multiple different extensions format. Partially for allowing multiple to be implemented by the same impl, but more so that you don't accidentally get different semantics because a different format was expected -- need good error handling w/ bad query
+
+| Parameter    | Type             | APIs       | Description        |
+| -----------  | ---------------- | ---------- | ---------------------- |
+| fields | Fields | Placeholder parameter for API Fields Extension. |
+| sort | Sort | Placeholder parameter for API Sort Extensions. |
+| query_type | string | Placeholder parameter for API Query Extension Type (e.g., `cql`). |
+| query | Query | Placeholder parameter for API Query Extension. |
+| operationName | string | Placeholder for GraphQL operationName parameter. |
+| variables | string | Placeholder for GraphQL variables parameter. |
+
+
+
+### Fields fields
+
+This object describes a Fields Extension format.
+
+| Field Name | Type  | Description |
+| ---------- | --------- | ----------- |
+| type       | string    | the Fields Extension name that is used for this, e.g., `include_exclude_csv` |
+| value      | object    | an extension-specific object representing the field filter |
+
+**type** the type as described in the Fields Extension 
+
+**value** the object representing whatever a field definitions means as specified by `type`.  This is completely implementation-specific.
+
+Example:
+
+```
+{ 
+  ...
+  "fields": {
+    "type": "include_exclude_array",
+    "value": {
+      "description": ["id","properties.eo:cloud_cover","properties.proj:geometry","properties.proj:crs"]
+    }
+  }
+  ...
+}
+```
+
+```
+{ 
+  ...
+  "fields": {
+    "type": "include_exclude_csv",
+    "value": {
+      "description": "@summary,-geometry"
+    }
+  }
+  ...
+}
+```
+
+
+### Query fields
+
+This object describes a Query Extension format.
+
+| Field Name | Type  | Description |
+| ---------- | --------- | ----------- |
+| query_type       | string    | the Query Extension name that is used for this, e.g., `cql` |
+| query      | object    | an extension-specific object representing the query filter |
+
+**type** the type as described in the Query Extension 
+
+**value** the object representing whatever a query means in the Query Extension specified by `type`.  This is completely implemenation-specific.
+
+Query Extensions are **recommended** to use attribute names qualified from the root of Item, rather than Item Properties.
+
+Example:
+
+```
+{ 
+  ...
+  "query_type": "cql",
+  "query": {
```

Comment:
How would this work for GET?  It seems like you'd still have to provide a json value in the query parameter, eg `?query={\"predicate\":\"xxx\"}` or use the dot notation `?query.predicate=xxx`.